### ISSUE TITLE
T5 Masked LM -- pre-trained model import?

### ISSUE THREAD

**Issue by moscow25** [huggingface/transformers#2514]
## ❓ Questions & Help

Hi, thanks for merging the T5 model! However it is not clear to me how to use the pretrained model for masked language modeling. It appears that the model example only returns a hidden state, or `T5WithLMHeadModel` which is not clear what this is doing -- it tends to return the same token for me in all locations. 

My understanding of the T5 paper was that one could add input tags like `<extra_id_0>` and receive multi-token masked responses in the decoder. Has this functionality been replicated in the codebase? And if not, do you think it possible to add it -- or do you have pointers to the community to try to add this ourselves?

Unfortunately, the T5 documentation for this use case is also not great. Which is too bad because multi-token masked responses seems like a great feature of the T5 model. Testing and inference for masked language modeling is simple for BERT and variants, but does not support multi-token responses.
